analytics2/utils/block_stats.py

```diff
@@ -198,6 +199,73 @@ def add_block_pop_density(bldg_pop: gpd.GeoDataFrame,
     return bldg_pop  
 
 
+def add_block_gini(bldg_pop: gpd.GeoDataFrame,
+                   values_col: str,
+                   block: Union[gpd.GeoDataFrame, str] = None,
+                   weights: libpysal.weights.weights.W = None
+                   ) -> gpd.GeoDataFrame:
+    """
+    Calculates the gini coefficient for the passed in values_col based on the passed in gpd.GeoDataFrame.
+    """
+    if 'block_id' not in bldg_pop.columns:
```

Comment:
i see [this code repeated a lot](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) - can we condense to a function and refactor, or ensure the incoming blocks do have `block_id` in their

Reply:
i think it's fine to document this in the docstrings and make it clear to the client that `block_id` needs to be in the columns instead of checking in _every_ feature generation function

Reply:
the client would then be responsible for calling `flex_load` before adding features